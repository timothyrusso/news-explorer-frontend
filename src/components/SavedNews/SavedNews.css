.saved-news {
  background-color: #F5F6F7;
  padding: 62px 0;
  display: flex;
  flex-direction: column;
}

@media screen and (max-width: 768px) {
  .saved-news {
    padding: 32px 0;
  }
}

@media screen and (max-width: 320px) {
  .saved-news {
    padding: 24px 0 13px;
  }
}
