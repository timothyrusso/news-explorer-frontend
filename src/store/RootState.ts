import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
