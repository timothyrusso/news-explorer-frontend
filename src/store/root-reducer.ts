import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
