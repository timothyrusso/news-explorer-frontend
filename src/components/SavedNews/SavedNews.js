import "./SavedNews.css";
import NewsCard from "../NewsCard/NewsCard";

const SavedNews = ({
  cards,
  savedCard,
  handleDeleteArticles,
  checkSavedArticle,
  saveUnauthorizedUserCard,
}) => {
  return (
    <section className="saved-news">
      <ul className="search-result-grid">
        {cards.map((card) => (
          <NewsCard
            key={card._id}
            card={card}
            savedCard={savedCard}
            handleDeleteArticles={handleDeleteArticles}
            checkSavedArticle={checkSavedArticle}
            saveUnauthorizedUserCard={saveUnauthorizedUserCard}
          />
        ))}
      </ul>
    </section>
  );
};

export default SavedNews;
