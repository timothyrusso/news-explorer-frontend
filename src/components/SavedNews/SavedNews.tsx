import React from 'react';
import NewsCard from '../NewsCard/NewsCard';
import { useSelector } from 'react-redux';
import { RootState } from '../../store/RootState';
import { SavedArticle } from '../../store/article/article.type';
import './SavedNews.css';

const SavedNews = () => {
  const savedArticles = useSelector(
    (state: RootState) => state.article.savedArticles
  );

  return (
    <section className="saved-news">
      <ul className="search-result-grid">
        {savedArticles.map((card: SavedArticle) => (
          <NewsCard key={card._id} card={card} />
        ))}
      </ul>
    </section>
  );
};

export default SavedNews;
