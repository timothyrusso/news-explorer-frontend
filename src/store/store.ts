import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
