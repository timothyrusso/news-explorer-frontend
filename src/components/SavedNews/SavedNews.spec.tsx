import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import SavedNews from './SavedNews';

describe('SavedNews component', () => {
  it('Should render the SavedNews component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SavedNews />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
