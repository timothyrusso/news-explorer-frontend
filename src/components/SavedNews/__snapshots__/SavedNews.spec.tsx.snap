// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedNews component Should render the SavedNews component 1`] = `
<DocumentFragment>
  <section
    class="saved-news"
  >
    <ul
      class="search-result-grid"
    />
  </section>
</DocumentFragment>
`;
